FirebaseAuth/Sources/Auth/FIRAuth.m

```diff
@@ -735,13 +741,90 @@ - (void)signInWithEmail:(NSString *)email
       [[FIRVerifyPasswordRequest alloc] initWithEmail:email
                                              password:password
                                  requestConfiguration:_requestConfiguration];
-
   if (![request.password length]) {
     callback(nil, [FIRAuthErrorUtils wrongPasswordErrorWithMessage:nil]);
     return;
   }
+#if TARGET_OS_IOS && !TARGET_OS_MACCATALYST
```

Comment:
Also needs to turn off visionOS like https://github.com/firebase/firebase-ios-sdk/pull/11483/files#diff-3d1f02c4c72f46392e059eea63284c3be76f20e3ee6da0a656a6a9bec570afe6R423